aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -397,58 +420,34 @@ std::tuple<Tensor &,Tensor &> mode_out(const Tensor& self, int64_t dim, bool kee
   }
 }
 
-std::tuple<Tensor, Tensor> max(const Tensor& self, int64_t dim, bool keepdim) {
-    Tensor max_indices = at::empty({0}, self.options().dtype(kLong));
-    if (self.is_quantized()) {
-      Tensor max = at::empty({0}, self.options().dtype(toUnderlying(self.scalar_type())));
-      at::native::max_out(self.int_repr(), dim, keepdim, max, max_indices);
-      // TODO: qscheme
-      return std::tuple<Tensor, Tensor>(at::_make_per_tensor_quantized_tensor(max,
-        self.q_scale(), self.q_zero_point()), max_indices);
+TORCH_IMPL_FUNC(max_out)
+(const Tensor& self,
+ int64_t dim,
+ bool keepdim,
+ const Tensor& values,
+ const Tensor& indices) {
+  NoNamesGuard guard;
+  if (self.numel() > 0) {
+    if (self.numel() == 1 && self.dim() == 0) {
+      values.fill_(self);
+      indices.fill_(0);
     } else {
-      Tensor max = at::empty({0}, self.options());
-      return at::native::max_out(self, dim, keepdim, max, max_indices);
+      auto& values_mut = const_cast<Tensor&>(values);
+      auto& indices_mut = const_cast<Tensor&>(indices);
```

Comment:
Is there a reason to use `const_cast` here instead of updating the types of `max_stub` (and the corresponding `max_kernel_impl` code in `TensorCompareKernel.cpp`) to use `const Tensor&` everywhere?

(If updating that signature requires changing a ton of call-sites then the const_cast is probably fine)